File: 4. Binary Tree/8_tree_value_count.java
class Node<T> {
    T val;
    Node<T> left;
    Node<T> right;

    public Node(T val) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

class Source {
    public static <T> int treeValueCount(Node<T> root, T target) {

        if (root == null) {
            return 0;
        }

        int currentOcc = root.val == target ? 1 : 0;

        currentOcc = currentOcc + treeValueCount(root.left, target);
        currentOcc = currentOcc + treeValueCount(root.right, target);

        return currentOcc;
    }

}

File: 4. Binary Tree/8_tree_value_count.py
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

def tree_value_count(root, target):
    if root is None:
        return 0

    current_occ = 1 if root.val == target else 0
    current_occ += tree_value_count(root.left, target)
    current_occ += tree_value_count(root.right, target)

    return current_occ
